components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java

```diff
@@ -304,4 +304,54 @@ public static boolean isFilteringEnhancementsEnabled() {
         return Boolean.parseBoolean(IdentityUtil.getProperty
                 (SCIMCommonConstants.SCIM_ENABLE_FILTERING_ENHANCEMENTS));
     }
+
+    /**
+     * Checks whether the identity.xml config is available for applying filtering enhancements. This makes sure that Eq
+     * should strictly check for the String match (in this case cross user store search should not be performed).
```

Comment:
Shall we simplify this method description a bit? the second sentence looks a bit complex to understand. This config will filter only from the primary domain if the domain is not explicitly present?

Reply:
Fixed